### DISCUSSION TITLE
Add a way to change the dimensions of a sparse matrix

### DISCUSSION THREAD

**Discussion by kaba2** [mathnet/mathnet-numerics#785] [Ideas]
Add a way to change the dimensions of a sparse matrix. This is a cheap operation for a matrix in compressed sparse row format. Currently it seems that you need to copy the matrix just to get more rows and columns, even though almost nothing changes in the internal data.

**Comment by cdrnet** [comment]
Yes, currently we consider the outer shape to be immutable. But we can certainly discuss this.
